File: server/datasets/java_source-code/Multiple-G2/unique_50_graph_traversal.java
import java.util.*;
public class GraphTraversalService {
    public static void main(String[] args) {
        GraphTraversalService g = new GraphTraversalService();
        Map<Integer, List<Integer>> adj = new HashMap<>();
        adj.put(1, Arrays.asList(2, 3));
        g.bfs(adj, 1);
    }
    public List<Integer> bfs(Map<Integer, List<Integer>> adj, int start) {
        List<Integer> order = new ArrayList<>();
        Set<Integer> visited = new HashSet<>();
        Queue<Integer> queue = new LinkedList<>();
        visited.add(start);
        queue.add(start);
        while (!queue.isEmpty()) {
            int node = queue.poll();
            order.add(node);
            for (int neighbor : adj.getOrDefault(node, Collections.emptyList())) {
                if (!visited.contains(neighbor)) {
                    visited.add(neighbor);
                    queue.add(neighbor);
                }
            }
        }
        return order;
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g2/unique_50_graph_traversal.py
from collections import deque

def breadth_first_search(adj_list, start_node):
    visited = set([start_node])
    queue = deque([start_node])
    order = []
    while queue:
        node = queue.popleft()
        order.append(node)
        for neighbor in adj_list.get(node, []):
            if neighbor not in visited:
                visited.add(neighbor)
                queue.append(neighbor)
    return order

def depth_first_search(adj_list, start_node):
    visited = set()
    order = []
    def dfs_helper(node):
        visited.add(node)
        order.append(node)
        for neighbor in adj_list.get(node, []):
            if neighbor not in visited:
                dfs_helper(neighbor)
    dfs_helper(start_node)
    return order
